src/main/scala/com/fulcrumgenomics/alignment/Aligner.scala

```diff
@@ -75,14 +75,30 @@ object Aligner {
   /** The minimum score allowed to start an alignment.  This prevents underflow. */
   val MinStartScore: Int = Int.MinValue / 2
 
+  /** Specialized matrix for storing integers that is markedly faster than using LinearMatrix[Int]. */
+  final class SimpleIntMatrix(val x: Int, val y: Int) extends Matrix[Int] {
+    private val data = new Array[Array[Int]](x)
+    forloop (from=0, until=x) { i => data(i) = new Array[Int](y) }
+
+    override def apply(i: Int, j: Int): Int = data(i)(j)
+    override def update(i: Int, j: Int, value: Int): Unit = data(i)(j) = value
+    override def dimensions: (Int, Int) = (x, y)
+  }
```

Comment:
I'm going to assume there's an implied `in` after `this` in your question.  I was 50/50 on doing that.  Mostly because I don't want us to forget why we specialized it and then re-generify it.  Maybe I'll move it over and add some verbose comments.